lib/spack/spack/build_systems/cmake.py

```diff
@@ -154,6 +156,44 @@ def _std_args(pkg):
         args.append('-DCMAKE_PREFIX_PATH:STRING={0}'.format(';'.join(deps)))
         return args
 
+    def flags_to_build_system_args(self, flags):
+        """Produces a list of all command line arguments to pass the specified
+        compiler flags to cmake. Note CMAKE does not have a cppflags option,
+        so cppflags will be added to cflags, cxxflags, and fflags to mimic the
+        behavior in other tools."""
+        # Has to be dynamic attribute due to caching
+        setattr(self, 'cmake_flag_args', [])
+
+        flag_string = '-DCMAKE_{0}_FLAGS={1}'
+        langs = {'C': 'c', 'CXX': 'cxx', 'Fortran': 'f'}
+
+        # Handle language compiler flags
+        for lang, pre in langs.items():
+            flag = pre + 'flags'
+            # cmake has no explicit cppflags support -> add it to all langs
+            lang_flags = ' '.join(flags.get(flag, []) + flags.get('cppflags',
+                                                                  []))
+            if lang_flags:
+                self.cmake_flag_args.append(flag_string.format(lang,
+                                                               lang_flags))
+
+        # Cmake has different linker arguments for different build types.
+        # We specify for each of them.
+        if flags['ldflags']:
+            ldflags = ' '.join(flags['ldflags'])
+            ld_string = '-DCMAKE_{0}_LINKER_FLAGS={1}'
```

Comment:
For multiple flags should this be enclosing in quotes to produce something like:

`-DCMAKE_..._LINKER_FLAGS='-a -b'`

? Likewise for all other constructions of environment/build-system variables

Reply:
I don't believe so. When you invoke cmake by hand, the quote marks are stripped by the shell and the multiple flags are passed as a single space-delimited string. I believe the python implementation is still passing them as a single string, because the underlying `Executable` implementation does not use the `shell` argument to `subprocess.Popen`.

If you're not convinced by this I'll take that as evidence that I shouldn't be so sure of myself and do some more rigorous testing of this.